packages/firestore/test/integration/api/validation.test.ts

```diff
@@ -795,6 +796,64 @@ apiDescribe('Validation:', persistence => {
       });
     });
 
+    validationIt(
+      persistence,
+      'cannot be sorted by an uncommitted server timestamp',
+      db => {
+        return withTestCollection(
+          persistence,
+          /*docs=*/ {},
+          async (collection: firestore.CollectionReference) => {
+            await db.disableNetwork();
+
+            const offlineAccumulator = new EventsAccumulator<
+              firestore.QuerySnapshot
+            >();
+            const onlineAccumulator = new EventsAccumulator<
+              firestore.QuerySnapshot
+            >();
```

Comment:
This is a slightly unconventional use of EventsAccumulator.  Normally we unconditionally accumulate events without any validation (usually by making it the direct receiver of the snapshot events) and then we call awaitEvent() to get the events out and then validate them.

I think what you're looking for is probably a Deferred. Take a look at https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/test/integration/api/database.test.ts#L700 for an example.

Reply:
Yeah, that's better; done.